public_html/wp-content/plugins/wordcamp-speaker-feedback/wordcamp-speaker-feedback.php

```diff
@@ -16,13 +16,92 @@
 
 define( __NAMESPACE__ . '\PLUGIN_DIR', \plugin_dir_path( __FILE__ ) );
 define( __NAMESPACE__ . '\PLUGIN_URL', \plugins_url( '/', __FILE__ ) );
+define( __NAMESPACE__ . '\OPTION_KEY', 'sft_feedback_page' );
 
+register_activation_hook( __FILE__, __NAMESPACE__ . '\activate' );
+register_deactivation_hook( __FILE__, __NAMESPACE__ . '\deactivate' );
 add_action( 'plugins_loaded', __NAMESPACE__ . '\load' );
 
+// Check if the page exists, and add it if not.
+add_action( 'init', __NAMESPACE__ . '\add_feedback_page' );
+
 /**
  * Include the rest of the plugin.
  */
 function load() {
 	require_once PLUGIN_DIR . 'includes/class-feedback.php';
 	require_once PLUGIN_DIR . 'includes/comment.php';
+	require_once PLUGIN_DIR . 'includes/form.php';
+	require_once PLUGIN_DIR . 'includes/page.php';
+}
+
+/**
+ * Create main Feedback page.
+ */
+function activate() {
+	add_feedback_page();
+}
+
+/**
+ * Create the Feedback page, save ID into an option.
+ */
+function add_feedback_page() {
+	$page_id = get_option( OPTION_KEY );
+	if ( $page_id ) {
+		return;
+	}
```

Comment:
It might be a good idea check and make sure an actual page is attached to this ID. Although, if the option gets deleted when the plugin is deactivated (per my other suggestion), that might eliminate any edge cases where the option still exists but the page doesn't...

I guess if this is running on every page load on `init` having the extra query for a specific page isn't great...

Reply:
I've updated it so that when the page is deleted, the option is deleted too — so we _shouldn't_ run into a case where the option still exists for a missing page.